be/src/vec/functions/array/function_array_union.cpp

```diff
@@ -0,0 +1,67 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "vec/functions/array/function_array_binary.h"
+#include "vec/functions/array/function_array_set.h"
+
+#include "vec/functions/simple_function_factory.h"
+
+namespace doris::vectorized {
+
+struct NameArrayUnion {
+    static constexpr auto name = "array_union";
+};
+
+template <typename Set, typename Element>
+struct UnionAction {
+    // True if set has null element
+    bool null_flag = false;
+    // True if result_set has null element
+    bool result_null_flag = false;
+    // True if it should apply the left array first.
+    static constexpr auto apply_left_first = true;
```

Comment:
apply_left_first  maybe rename to execute_left_column_first or process_left_column_first, will be better.